src/app/tasks/project-tasks-list/project-tasks-list.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, OnInit, Inject, Input, Output, EventEmitter } from '@angular/core';
+import { taskService, groupService, gradeService } from 'src/app/ajs-upgraded-providers';
+
+@Component({
+  selector: 'df-project-tasks-list',
+  templateUrl: './project-tasks-list.component.html',
+  styleUrls: ['./project-tasks-list.component.scss'],
+})
+export class ProjectTasksListComponent implements OnInit {
+  @Input() unit: any;
+  @Input() project: any = {
+    tasks: []
+  };
+
+  @Output() private onSelect = new EventEmitter<any>();
+
+  constructor(
+    @Inject(taskService) private ts: any,
+    @Inject(groupService) private grounds: any,
+    @Inject(gradeService) private grades: any
+  ) {}
+
+  ngOnInit(): void {}
+
+  get statusText(): string {
+    return this.ts.statusText;
+  }
+
+  get hideGroupSetName(): boolean {
+    return this.unit.group_sets.length === 0;
+  }
+
+  public groupSetName(id): string {
+    return this.grounds.groupSetName(id, this.unit);
+  }
```

Comment:
Are any of these being used? I couldn't find anything that was calling them

Reply:
If you need to use these functions I think the best way is init the variable then assign values in ` ngoninit` function. (You are not using ` ngoninit` function so need to remove that as well if not functional). If the value will change try to find out if the ` ts.statusText` is an observerable so you can subscribe to that observerable object. Also `ts` is not a good way to name a variable should be clear like `taskService`. Same for ` groups` and `grades` should change to ` groupservices` and ` gradeservices`. ` groups ` sounds like an array contain all the group.